### PR TITLE
Parser exception

### PR DESCRIPTION
**Pull request by tarberd**

### CODE REVIEW

**Review by renannetto — COMMENTED** [review]
Is there an example using this?

**Comment by tarberd** [comment]
@renannetto 
FILE* ifp = fopen(filename.c_str(), "r");
if(!ipf)

**Comment by tarberd** [comment]
@renannetto 
FILE\* ifp = fopen(filename.c_str(), "r");
if(!ifp){ throw InexistentFile(); }

File: ophidian/parser/ParserException.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef PARSER_EXCEPTION_H
+#define PARSER_EXCEPTION_H
+
+#include <string>
+
```
Comment (csguth) [inline ophidian/parser/ParserException.h:5]: `#include <exception>`

File: ophidian/parser/ParserException.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace ophidian {
+    namespace parser {
+
+        class InexistentFile {
```
Comment (csguth) [inline ophidian/parser/ParserException.h]: `class InexistentFile : public std::exception {`

File: ophidian/parser/ParserException.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            std::string errorMessage(){
```
Comment (csguth) [inline ophidian/parser/ParserException.h]:
```
const char* what() const override {
   return "The given file was not found";
}
```

**Review by csguth — CHANGES_REQUESTED** [review]

**Comment by tarberd** [comment]
@csguth now it's following the standards ;)

File: ophidian/parser/ParserException.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#ifndef PARSER_EXCEPTION_H
+#define PARSER_EXCEPTION_H
+
+#include <exception>
+
+namespace ophidian {
+    namespace parser {
+
+        class InexistentFile : public std::exception {
+            const char* what() override {
```
Comment (csguth) [inline ophidian/parser/ParserException.h]: please, make it public

**Review by csguth — CHANGES_REQUESTED** [review]

**Comment by renannetto** [comment]
can you also add a test that throws this exception when parsing an invalid file?

**Comment by csguth** [comment]
@tarberd Pease, update this branch. Otherwise I can't merge it.

**Review by csguth — APPROVED** [review]

**Pull request merged**